docs/blog/2025-01-21-project-references.md

```diff
@@ -0,0 +1,177 @@
+---
+title: What Are TypeScript Project References?
+slug: typescript-project-references
+authors: [Zack DeRose]
+tags: [typescript, monorepo, nx]
+cover_image: /blog/images/2025-01-21/ts-islands.png
+---
+
+## Connecting Projects At The TypeScript Level
+
+Consider the following workspace:
+
+```filesystem
+is-even
+  index.ts
+  tsconfig.json
+is-odd
+  index.ts
+  tsconfig.json
+tsconfig.base.json
+```
+
+If we try to run our build script on the `is-odd` project, we see that TypeScript is unable to run the `tsc` command because at the TypeScript level, `is-odd` is not aware of the `is-even` module:
+
+```shell
+ . > cd is-odd
```

Comment:
Instead of "cd" into the directory which I think is not relevant here, you should be able to pass a "path" property:
https://github.com/nrwl/nx/tree/master/docs#terminal-output

Like:

‎``` {% command="tsc" path="~/is-odd" %}
index.ts:1:24 - error TS2792: Cannot find module 'is-even'. Did you mean to set the 'moduleResolution' option to 'nodenext', or to add aliases to the 'paths' option?

1 import { isEven } from 'is-even';
                         ~~~~~~~~~

Found 1 error in index.ts:1
‎```